### PR TITLE
fix: should support module.exports = function*(ctx) {} as a controller

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

closes https://github.com/eggjs/egg/issues/785

### CODE REVIEW

**Comment by ngot** [comment]
+1

**Comment by atian25** [comment]
覆盖率降了，少了个分支

**Comment by popomore** [comment]
else 没走到

File: test/fixtures/controller-app/app/controller/number.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+'use strict';
+
+module.exports = 123;
```
Comment (atian25) [inline test/fixtures/controller-app/app/controller/number.js]: 这种类型的是不是得报错

Comment (dead-horse) [inline test/fixtures/controller-app/app/controller/number.js]: 现在不能报错，会不兼容

Comment (dead-horse) [inline test/fixtures/controller-app/app/controller/number.js]: 只有将他用作一个 controller 的时候才会报错

Comment (atian25) [inline test/fixtures/controller-app/app/controller/number.js]: 合吧

**Comment by popomore** [comment]
+ egg-core@3.4.1

**Pull request merged**